src/priority-queue-original.js

```diff
@@ -0,0 +1,60 @@
+/**
+ * A priority queue stores a list of items but each can have a numeric priority value.
+ * Items with a higher priority are dequeued before items with a lower priority.
+ * Implemented as a hash of arrays where the hash keys are priority values.
+ */
+function PriorityQueue(size) {
+	this.store = {};	// keys are priorities, values are arrays of elements
+	this.count = 0;
+
+	// adds an item
+	// priority must be an integer (higher value has higher priority)
+	this.add = function(value, priority) {
+		if (this.store[priority] == undefined)
+			this.store[priority] = [];
+
+		this.store[priority].push(value);
+		this.count++;
+	};
+
+	// returns the oldest-added value with the highest priority
+	this.Pop = function() {
+		maxKey = Math.max(Object.keys(this.store));
+		this.count--;
+		return this.store[maxKey].shift();
+	};
+
+	this.length = function() {
+		return this.count;
+	}
+}
+
+PriorityQueue.prototype.get_all_priorities = function() {
+	return Object.keys(this.store);
+}
+
+// iterates through all the queue elements in priority-then-FIFO order
+PriorityQueue.prototype.forEach = function(callback) {
+	var keys = Object.keys(this.store).sort();
+
+	for (var a = keys.length; a > 0; a--) {
+		for (var b = 0; b < this.store[a].length; b++)
+			callback(this.store[a][b]);
+	}
```

Comment:
```suggestion
	var priorities = Object.keys(this.store).sort();

	for (var priority = priorities.length; priority > 0; priority--) {
		for (var i = 0; i < this.store[priority].length; i++)
			callback(this.store[priority][i]);
	}
```
Don't commit as-is, but just to illustrate choosing more meaningful variable names (even `i` says something). IMO, it's now much clearer what this code is trying to do.